### PR TITLE
Feature/revamp model endpoints

### PR DESCRIPTION
**Pull request by AlexejPenner**
## Describe changes
I made model_versions their own root route - for the purpose of preserving some of the implicit functionality of model versions I left two endpoints on the /models route to allow users to fetch model versions based on latest, stage or name

image

We need to have all the model_version endpoints with a root level to allow for all sort of operations through the dashboard.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by AlexejPenner** [comment]
@avishniakov please let me know if I am missing anything obvious - I have opted to keep the behaviour for get and list methods consistent, but everything else now relies on the new more atomic interface

File: src/zenml/zen_server/routers/model_versions_endpoints.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    return zen_store().list_model_version_artifact_links(
+        model_version_id=model_version_id,
+        model_version_artifact_link_filter_model=model_version_artifact_link_filter_model,
+    )
+
+
+@router.delete(
+    "/{model_version_id}"
+    + ARTIFACTS
+    + "{model_version_artifact_link_name_or_id}",
```
Comment (avishniakov) [inline src/zenml/zen_server/routers/model_versions_endpoints.py:191]:
```suggestion
    + "/{model_version_artifact_link_name_or_id}",
```

**Review by avishniakov — APPROVED** [review]
One slash missing, other than that - if tests pass, all good.

**Review by avishniakov — CHANGES_REQUESTED** [review]
Fix status, cause of slash :)

**Comment by AlexejPenner** [comment]
@avishniakov do you have any manual testing flow that you used, that I can do to ensure no functionality is lost?

**Comment by avishniakov** [comment]
> @avishniakov do you have any manual testing flow that you used, that I can do to ensure no functionality is lost?

I think, you can run the E2E template on a remote (docker) setup - should be good enough. Otherwise, it is pretty well covered with tests already.

**Review by fa9r — COMMENTED** [review]
A lot of the changes introduced here clash with #2044, please consolidate this first

File: src/zenml/models/model_models.py [hunk 2]
```diff
@@ -237,6 +237,9 @@ def to_model_version(
             was_created_in_this_run=was_created_in_this_run,
             suppress_class_validation_warnings=suppress_class_validation_warnings,
         )
+        mv._id = self.id
+
+        return mv
```
Comment (AlexejPenner) [inline src/zenml/models/model_models.py:242]: @avishniakov I discovered this value was never set, i implemented a temporary fix but would leave it up to you to decide where to handle this

**Pull request merged**